crates/ibc/src/core/router.rs

```diff
@@ -26,39 +23,7 @@ pub trait Router {
     fn get_route_mut(&mut self, module_id: &ModuleId) -> Option<&mut dyn Module>;
 
     /// Return the module_id associated with a given port_id
-    fn lookup_module_by_port(&self, port_id: &PortId) -> Option<ModuleId>;
-
-    fn lookup_module_channel(&self, msg: &ChannelMsg) -> Result<ModuleId, ChannelError> {
-        let port_id = match msg {
-            ChannelMsg::OpenInit(msg) => &msg.port_id_on_a,
-            ChannelMsg::OpenTry(msg) => &msg.port_id_on_b,
-            ChannelMsg::OpenAck(msg) => &msg.port_id_on_a,
-            ChannelMsg::OpenConfirm(msg) => &msg.port_id_on_b,
-            ChannelMsg::CloseInit(msg) => &msg.port_id_on_a,
-            ChannelMsg::CloseConfirm(msg) => &msg.port_id_on_b,
-        };
-        let module_id = self
-            .lookup_module_by_port(port_id)
-            .ok_or(ChannelError::Port(UnknownPort {
-                port_id: port_id.clone(),
-            }))?;
-        Ok(module_id)
-    }
-
-    fn lookup_module_packet(&self, msg: &PacketMsg) -> Result<ModuleId, ChannelError> {
-        let port_id = match msg {
-            PacketMsg::Recv(msg) => &msg.packet.port_id_on_b,
-            PacketMsg::Ack(msg) => &msg.packet.port_id_on_a,
-            PacketMsg::Timeout(msg) => &msg.packet.port_id_on_a,
-            PacketMsg::TimeoutOnClose(msg) => &msg.packet.port_id_on_a,
-        };
-        let module_id = self
-            .lookup_module_by_port(port_id)
-            .ok_or(ChannelError::Port(UnknownPort {
-                port_id: port_id.clone(),
-            }))?;
-        Ok(module_id)
-    }
+    fn lookup_module(&self, port_id: &PortId) -> Option<ModuleId>;
```

Comment:
While we're at it, do we even need the `ModuleId`? Does it provide any value? We could simply have

```rust
pub trait Router {
    fn get_route(&self, port_id: &PortId) -> Option<&dyn Module>;
    fn get_route_mut(&mut self, port_id: &PortId) -> Option<&mut dyn Module>;
}
```

Reply:
Good Point!
The only noteworthy benefit rn is more efficiency in searching for modules compared to directly using `port_id`. 
Ran a local test on `MockContext`... With `ModuleId` it can be up to ~10 times faster at times. HW should say we are talking of ~1 microsecond.
Anyway, for now, I dropped `ModuleId` ([9227ef7](https://github.com/cosmos/ibc-rs/pull/806/commits/9227ef7557aa8437b4eec5dc6fd1311fa1af0333)). It can be added at any time if it would really matter.

Reply:
Ahhh, I just recalled something. Reverted the above commit!
This `ModuleId` thing is essential for apps like ics23. Upon registering each interchain account on the controller chain a new `PortId` will be allocated at a certain height. As a result, there could be a bunch of `PortId`s instantiated. This `module_lookup` efficiently maps them to a single module instance. Otherwise, for each of those `PortId`s, a separate redundant module instance should be stored.

Reply:
Great, let's not do that then!